lib/sprockets/sass/importer.rb

```diff
@@ -30,65 +35,113 @@ def mtime(path, options)
-          :importer => self
+          filename: pathname.to_s,
+          syntax: syntax(pathname),
+          importer: self,
+          custom: { sprockets_context: context }
         )
       end
 
       # Create a Sass::Engine that will handle importing
       # a glob of files.
       def engine_from_glob(glob, base_path, options)
         context = options[:custom][:sprockets_context]
-        imports = resolve_glob(context, glob, base_path).inject('') do |imports, path|
+        engine_imports = resolve_glob(context, glob, base_path).reduce('') do |imports, path|
           context.depend_on path
           relative_path = path.relative_path_from Pathname.new(base_path).dirname
           imports << %(@import "#{relative_path}";\n)
         end
-        return nil if imports.empty?
-        ::Sass::Engine.new imports, options.merge(
-          :filename => base_path.to_s,
-          :syntax   => :scss,
-          :importer => self
+        return nil if engine_imports.empty?
+        ::Sass::Engine.new engine_imports, options.merge(
+          filename: base_path.to_s,
+          syntax: syntax(base_path.to_s),
+          importer: self,
+          custom: { sprockets_context: context }
         )
       end
 
+      def resolve_path_with_load_paths(context, path, root_path, file)
+        context.resolve(file.to_s, load_paths: context.environment.paths, base_path: root_path, accept: syntax_mime_type(path))
+      rescue
+        nil
+      end
+
       # Finds an asset from the given path. This is where
       # we make Sprockets behave like Sass, and import partial
       # style paths.
       def resolve(context, path, base_path)
-        possible_files(context, path, base_path).each do |file|
-          context.resolve(file) { |found| return found if context.asset_requirable?(found) }
+        paths, root_path = possible_files(context, path, base_path)
+        if Sprockets::Sass.version_of_sprockets >= 3
+          paths.each do |file|
+            found_item = resolve_path_with_load_paths(context, path, root_path, file)
+            return found_item if !found_item.nil? && asset_requirable?(context, found_item)
+          end
+        else
+          paths.each do |file|
+            context.resolve(file.to_s) do |found|
+              return found if context.asset_requirable?(found)
+            end
+          end
         end
 
         nil
       end
 
+      def available_content_types(path)
+        ['text/css', syntax_mime_type(path), "text/#{syntax(path)}+ruby"]
+      end
+
+      def resolve_path(context, path)
+        context.resolve(path.to_s)
+      rescue
+        nil
+      end
+
+      def stat_of_pathname(context, pathname, path)
+        if Sprockets::Sass.version_of_sprockets >= 4
+          asset = context.environment.load(pathname)
+          context.environment.stat(asset.filename)
+        else
+          context.environment.stat(path)
+        end
+      end
+
+      def asset_requirable?(context, path)
```

Comment:
I am not sure, the method asset_requirable? i added tries to mimic the old method from sprockets 2.x . This was the original implementation which was done against the ```context``` variable.

```
 # Tests if target path is able to be safely required into the
    # current concatenation.
    def asset_requirable?(path)
      pathname = resolve(path)
      content_type = environment.content_type_of(pathname)
      stat = environment.stat(path)
      return false unless stat && stat.file?
      self.content_type.nil? || self.content_type == content_type
    end
```

But since the new implementation needs to work with sprockets 4.x checking contex.content_type to be equal with the content type of path is not working due to the fact that it introduces new mime types.  So the only difference is the last line of the code.



So in order to work, i check against all available mime types . This is because context.content_type returns text/css but ```content_type_of_path``` will sometime return text/scss or text/scss+ruby , so is not really the same ..

Previously with sprockets 2.x that worked because was only 'text/css' as mime type used.